### DISCUSSION TITLE
How to enable aarch64 version of SPHINCS+ SHAKE algorithms?

### DISCUSSION THREAD

**Discussion by wadaphaq** [open-quantum-safe/liboqs#1965] [Q&A]
When testing the algorithms' performace with and without CPU extensions, I noticed that sphincs shake algorithms have same performance with or without the CPU extension enabled in the building process. Upon inspection of the liboqs I noticed that the cmakelists in the sphincs src folder do not include an aarch64 version, and even after building liboqs the aarch64 version is missing from the source folder. Is there a way to activate and build the aarch64 version of the sphincs shake algorithms?

**Comment by SWilson4** [comment]
Thanks for the question! It does look to me as though we never actually build the aarch64 versions. This is, of course, an oversight on our part. I'll create an issue to fix it.

**Reply by wadaphaq** [reply]
Hi, I have read some of the issues related to the SPHINCS+ and regarding how they are not enabled in the first place. I would like to help contribute to the issue! After some modification to the cmakelists and the src of sphincs, after building the liboqs, the aarch64 version appears in the build/src but does not use this version when running tests such as speed_sig.

Can you give me some pointers on how to enable it or correctly run this version of the algorithm when testing?

Thank you very much!

**Reply by SWilson4** [reply]
Hi @wadaphaq,

If you would like to work on pulling in the aarch64 implementation of SPHINCS+ from PQClean, I would suggest that you first familiarize yourself with our [`copy_from_upstream`](https://github.com/open-quantum-safe/liboqs/wiki/Patching-upstream-sources) process. This is the means by which source code is integrated (hopefully) seamlessly into the codebase—including automatic modifications to the necessary CMakeLists and `sig_sphincs_*.c` files so that the desired implementation is built and enabled.

If you find the documentation lacking or you get stuck, please don't be afraid to ask. This also helps us improve the quality of our docs and contributing guides.
